protocol/avdl/rekey.avdl

```diff
@@ -32,6 +38,12 @@ protocol rekey {
     array<ProblemTLF> tlfs;
   }
 
+  record ProblemSetDevices {
+    User user;
+    KID kid;
+    array<ProblemTLFDevices> tlfs;
```

Comment:
This is fine but more complicated than i expected.

would it have worked to just expose an map of KID -> Devices in addition to the prexisting ProblemSet?